### PR TITLE
Remove confirmation countdown timer

### LINKED ISSUE #144: Remove confirmation countdown timer
From work party -- remove confirmation countdown timer, user confirmation dialog. Change sign out as well to skip confirming all together.

### PR DESCRIPTION
**Pull request by juliusl**
- Removed countdown timer for action buttons. Instead display confirm…
- Add ConfirmationDialog component.
- Added noconfirmation option for action buttons
- Changed sign-out butotn to be a button that says sign out.

Note: This doesn't include all the layout changes that need to be made overall.

Fixes #144.

### CODE REVIEW

File: src/site/scripts/components/profile/actionbutton.tsx [hunk 1]
```diff
@@ -26,47 +22,32 @@ export interface IActionButtonProps {
      * Textual display.
      */
     text: string;
+
+    /**
+     * Confirmation text to display in the confirmation dialog.
+     */
+    confirmationText?: string;
```
Comment (JoshuaKGoldberg) [inline src/site/scripts/components/profile/actionbutton.tsx]: Nit: can you keep these in alphabetical order?

File: src/site/scripts/components/profile/actionbutton.tsx [hunk 2]
```diff
@@ -88,25 +69,46 @@ export class ActionButton extends React.Component<IActionButtonProps, IActionBut
...
      * @returns The rendered input button.
      */
-    private renderButton(): JSX.Element {
-        return (
-            <input
-                className="action-button"
-                onClick={(): void => this.toggleExpansion()}
-                type="button"
-                value={this.props.text} />);
+    private renderButton(shortCircuitAction: boolean): JSX.Element {
```
Comment (JoshuaKGoldberg) [inline src/site/scripts/components/profile/actionbutton.tsx]: JSDoc `shortCircuitAction`?

Also, a better name might be `skipAction`?

File: src/site/scripts/components/profile/actionbutton.tsx [hunk 3]
```diff
@@ -88,25 +69,46 @@ export class ActionButton extends React.Component<IActionButtonProps, IActionBut
...
+        if (shortCircuitAction) {
+            let onClick = (): void => {
```
Comment (JoshuaKGoldberg) [inline src/site/scripts/components/profile/actionbutton.tsx]: Why not declare `onClick` inline in the input?

File: src/site/scripts/components/profile/actionbutton.tsx [hunk 4]
```diff
@@ -88,25 +69,46 @@ export class ActionButton extends React.Component<IActionButtonProps, IActionBut
...
+                this.props.action();
+            };
+
+            return (
+                <input
+                    className="action-button"
+                    onClick={(onClick)}
```
Comment (JoshuaKGoldberg) [inline src/site/scripts/components/profile/actionbutton.tsx]: Unnecessary parenthesis

File: src/site/scripts/components/profile/actionbutton.tsx [hunk 5]
```diff
@@ -88,25 +69,46 @@ export class ActionButton extends React.Component<IActionButtonProps, IActionBut
...
+                    type="button"
+                    value={this.props.text} />);
+        } else {
```
Comment (JoshuaKGoldberg) [inline src/site/scripts/components/profile/actionbutton.tsx]: Remove else, reduce nesting.

File: src/site/scripts/components/profile/actionbutton.tsx (hunk 3 above)
Comment (JoshuaKGoldberg) [inline src/site/scripts/components/profile/actionbutton.tsx]: Even better: the only difference between the `input`s is the onClick. Instead of this big if/else, have one `const onClick` that's defined with a ternary.

File: src/site/scripts/components/profile/actionbutton.tsx [hunk 6]
```diff
@@ -116,77 +118,40 @@ export class ActionButton extends React.Component<IActionButtonProps, IActionBut
...
             <div className="action-confirmation">
-                <input
-                    onClick={onClick}
-                    type="button"
-                    value={value} />
-                    {this.renderCancel()}
+                <div className="action-confirmation-overlay"/>
+                <ConfirmationDialog
+                    action={() => onClick()}
+                    onCancel={() => this.toggleExpansion()}
```
Comment (JoshuaKGoldberg) [inline src/site/scripts/components/profile/actionbutton.tsx]: Nit: can you manually annotate `: void` in all these lambdas? I like this as a rule because it helps avoid accidentally triggering unexpected behavior when returning the value of functions to things that react to return values.

File: src/site/scripts/components/profile/actions.tsx [hunk 7]
```diff
@@ -14,6 +14,11 @@ export interface IActionProps {
     alive: boolean;
 
     /**
+     * The user's target
```
Comment (JoshuaKGoldberg) [inline src/site/scripts/components/profile/actions.tsx]: Nit: missing period.

File: src/site/scripts/components/profile/actions.tsx [hunk 8]
```diff
@@ -28,6 +33,7 @@ export interface IActionProps {
  * Component for actionable profile buttons.
  */
 export const Actions: React.StatelessComponent<IActionProps> = (props: IActionProps): JSX.Element => {
+    let killButton  = `I killed ${props.target}!`;
```
Comment (JoshuaKGoldberg) [inline src/site/scripts/components/profile/actions.tsx]: Why declare this if it' s only used once?

Also, nit: prefer `const` over `let`.

File: src/site/scripts/components/profile/confirmationdialog.tsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+export interface IConfirmationDialogProps {
+    /**
+     * The action to execute on confirm.
+     */
+    action: () => void;
+
+    /**
+     * The action to execute on cancel.
+     */
+    onCancel: () => void;
```
Comment (JoshuaKGoldberg) [inline src/site/scripts/components/profile/confirmationdialog.tsx]: Nit: alphabetical order?

File: src/site/scripts/components/profile/actions.tsx [hunk 10]
```diff
@@ -37,7 +43,7 @@ export const Actions: React.StatelessComponent<IActionProps> = (props: IActionPr
 
     return (
         <div className="actions actions-alive">
-            <ActionButton action={(): void => props.onKill()} text="Report a kill!" />
-            <ActionButton action={(): void => props.onDeath()} text="Are you dead?" />
+            <ActionButton action={(): void => props.onKill()} text={killButton} confirmationText="Are you sure you want to report a kill?"/>
+            <ActionButton action={(): void => props.onDeath()} text="I'm dead!" confirmationText="Are you sure you want to report yourself as dead?"/>
```
Comment (JoshuaKGoldberg) [inline src/site/scripts/components/profile/actions.tsx]: Nit: these lines are getting pretty wide. Can you split the attributes across lines?

File: src/site/scripts/components/profile/confirmationdialog.tsx [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+     * The confirmation text to display.
+     */
+    confirmationText: string;
+}
+
+/**
+ * A component for a confirmation dialog box.
+ */
+export const ConfirmationDialog: React.StatelessComponent<IConfirmationDialogProps> = (props: IConfirmationDialogProps): JSX.Element => {
+        return (
```
Comment (JoshuaKGoldberg) [inline src/site/scripts/components/profile/confirmationdialog.tsx]: Appears to be an extra level of indentation?

File: src/site/styles/layout.less [hunk 12]
```diff
@@ -99,15 +134,10 @@ input {
     }
 }
 
-#logout .action-x {
+#logout .action-Sign-Out {
```
Comment (JoshuaKGoldberg) [inline src/site/styles/layout.less]: Instead of having this ugly class name, how about lowercasing the class name in the component?

File: src/site/styles/variables.less [hunk 13]
```diff
@@ -7,10 +7,11 @@
 @color-shadow-medium: #b0c0c0;
 @color-shadow-heavy: #aabbbb;
 @color-shadow-dark: #212835;
+@color-accent-color: #329A97;
 
-@font-body-family: "Segoe UI Semilight", "Segoe UI", "sans-serif";
+@font-body-family: "Segoe UI Semilight", "Segoe UI", "Lucida Grande", "sans-serif";
 @font-body-size: 1.17em;
-@font-fancy-family: "Segoe UI Light";
+@font-fancy-family: "Segoe UI Light", "Lucida Grande Narrow Light", "sans-serif";
```
Comment (JoshuaKGoldberg) [inline src/site/styles/variables.less]: Why?

Comment (JoshuaKGoldberg) [inline src/site/styles/variables.less]: If it's for the font fallback issues,
1. That fix should be in its own PR
2. We want to have an actual font fallback for consistent fonts across browsers

File: src/site/styles/layout.less [hunk 14]
```diff
@@ -83,11 +83,46 @@ input {
...
+
+    .action-confirmation-container {
+        position: fixed;
+        left: 0;
+        right: 0;
+        margin: -15% auto;
+        background: white;
+        z-index: 300;
+        width: 500px;
+        height: 200px;
```
Comment (JoshuaKGoldberg) [inline src/site/styles/layout.less]: You're doing an awful lot of width and height setting. Try `display: flex` instead.

File: src/site/styles/layout.less [hunk 15]
```diff
@@ -83,11 +83,46 @@ input {
...
+            top: 30%;
+        }
+        
+        .action-confirmation-buttons {
+            display: table-row;
+            position: absolute;
+            bottom: 0;
+            margin-bottom: 12px;
+            input {
+                display: table-column;
```
Comment (JoshuaKGoldberg) [inline src/site/styles/layout.less]: Using the table-based `display` values probably isn't needed here (another plus for flex).

File: src/site/styles/variables.less (hunk 13 above)
Comment (JoshuaKGoldberg) [inline src/site/styles/variables.less]: I've since fixed the font fallback issues.

File: src/site/styles/layout.less [hunk 16]
```diff
@@ -83,11 +83,46 @@ input {
     position: relative;
 }
 
-.action-confirmation {
-    position: absolute;
-    top: 0;
-    left: 0;
-    background: white;
+.action-confirmation {
```
Comment (JoshuaKGoldberg) [inline src/site/styles/layout.less]: Why nest the rules under `.action-confirmation`? You don't need the extra specificity, it just adds CSS bloat.

File: src/site/scripts/components/profile/actionbutton.tsx (hunk 1 above)
Comment (juliusl) [inline src/site/scripts/components/profile/actionbutton.tsx]: For the sake of this file. Thumbs up means resolved. :)

File: src/site/scripts/components/profile/actionbutton.tsx (hunk 2 above)
Comment (juliusl) [inline src/site/scripts/components/profile/actionbutton.tsx]: I think `skipConfirm` is probably a more accurate description.

File: src/site/styles/variables.less (hunk 13 above)
Comment (juliusl) [inline src/site/styles/variables.less]: This is what Chelsea asked for.

File: src/site/styles/layout.less (hunk 14 above)
Comment (juliusl) [inline src/site/styles/layout.less]: Layout changes like that should be handled in a wholesale style PR instead of this sort of PR.

File: src/site/styles/layout.less (hunk 15 above)
Comment (juliusl) [inline src/site/styles/layout.less]: Layout changes like that should be handled in a wholesale style PR instead of this sort of PR.

File: src/site/styles/layout.less (hunk 16 above)
Comment (juliusl) [inline src/site/styles/layout.less]: More specificity is better, it adds the distinction that these rules are not to be used on other components and are specific to a single component.

**Comment by juliusl** [comment]
Rebased before updating
